.mint/ruby-binary-builder.yml

```diff
@@ -0,0 +1,121 @@
+### required init properties
+### init:
+###  ruby-version: string
+
+tasks:
+  - key: install-ruby
+    run: |
+      # install ruby install
+      set -ueo pipefail
+
+      sudo apt-get update
+
+      tmp_dir=$(mktemp -d)
+
+      function cleanup {
+        if [ -z "$tmp_dir" ]; then
+          rm -rf "$tmp_dir"
+        fi
+      }
+      trap cleanup EXIT
+
+      sudo apt-get install -y make
+
+      cd "$tmp_dir"
+      curl -fsSLO https://github.com/postmodern/ruby-install/releases/download/v0.9.2/ruby-install-0.9.2.tar.gz
+      tar -xzf ruby-install-0.9.2.tar.gz
+      cd ruby-install-0.9.2/
+      sudo make install
+
+      sudo apt-get clean
+
+      # install ruby
+      export XDG_CACHE_HOME="./tmp/cache"
+      source_dir="./tmp/src"
+
+      function cleanup {
+        # TODO: mintignore these instead of removing
+        echo "Cleaning up"
+        sudo rm -rf "$source_dir" "$XDG_CACHE_HOME"
+      }
+      trap cleanup EXIT
+
```

Comment:
I think this will overwrite the trap for EXIT defined on line 20.

Reply:
ope, you're right. I should've more closely looked at these when I merged them